src/main/java/com/iexec/core/sms/SmsService.java

```diff
@@ -55,28 +55,37 @@ public SmsService(PlatformRegistryConfiguration registryConfiguration,
      * If any of these conditions is wrong, then the {@link SmsClient} is considered to be not-ready.
      *
      * @param chainTaskId ID of the on-chain task.
-     * @param tag Tag of the deal
-     * @return {@literal true} if previous conditions are met, {@literal false} otherwise.
+     * @param tag Tag of the deal.
+     * @return SMS url if TEE types of tag & SMS match.
      */
-    public String getVerifiedSmsUrl(String chainTaskId, String tag) {
+    public Optional<String> getVerifiedSmsUrl(String chainTaskId, String tag) {
         final TeeEnclaveProvider teeEnclaveProviderForDeal = TeeUtils.getTeeEnclaveProvider(tag);
         if(teeEnclaveProviderForDeal == null){
-            return "";
+            log.error("Can't get verified SMS url with invalid TEE enclave " + 
+                "provider from tag [chainTaskId:{}]", chainTaskId);
+            return Optional.empty();
         }
-        String smsUrl = retrieveSmsUrl(teeEnclaveProviderForDeal);
-        final SmsClient smsClient = smsClientProvider.getSmsClient(smsUrl);
+        Optional<String> smsUrl = retrieveSmsUrl(teeEnclaveProviderForDeal);
+        if(smsUrl.isEmpty()){
+            log.error("Can't get verified SMS url since type of tag is not " + 
+                "supported [chainTaskId:{}]", chainTaskId);
```

Comment:
```suggestion
            log.error("Can't get verified SMS url since type of tag is not " + 
                "supported [chainTaskId:{},teeEnclaveProvider:{}]", chainTaskId, teeEnclaveProviderForDeal);
```